test/api/model/model_test.dart

```diff
@@ -212,4 +215,97 @@ void main() {
         .deepEquals([2, 3, 11]);
     });
   });
+
+  group('MessageEditState', () {
+    Map<String, dynamic> baseJson() => deepToJson(eg.streamMessage()) as Map<String, dynamic>;
+
+    test('edit history is absent', () {
+      check(Message.fromJson(baseJson()..['edit_history'] = null))
+        .editState.equals(MessageEditState.none);
+
+      check(Message.fromJson(baseJson()
+          ..['edit_history'] = null
+          ..['last_edit_timestamp'] = 1678139636))
+        .editState.equals(MessageEditState.edited);
+    });
+
+    void checkEditState(MessageEditState editState, List<Map<String, dynamic>> editHistory){
+      check(Message.fromJson(baseJson()..['edit_history'] = editHistory))
+        .editState.equals(editState);
+    }
+
+    group('edit history exists', () {
```

Comment:
Oh, I just noticed: it seems that a single item in the edit-history list could represent _both_ a content change and a move. From the [doc](https://zulip.com/api/get-messages):

> The other fields are optional, and will be present or not depending on whether the channel, topic, and/or message content were modified in the edit event.

So let's add a test that includes both a content change and a move in the same edit-history item. The move could be a topic change or a stream change, or we could have separate tests for each.

Reply:
Added the test "Both topic and content changed in a single edit -> edited".